In POSIX sh, jobs flags beside -l and -p are undefined

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related


#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
jobs -s
jobs -r
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
The only acceptable flags under POSIX sh for `jobs` are `-l` and `-p` ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/jobs.html)). The `-s` and `-r` flags are [valid on Bash](https://www.gnu.org/software/bash/manual/html_node/Job-Control-Builtins.html#Job-Control-Builtins) to check for stopped/suspended jobs and running jobs, but invalid on, for instance, dash:
```sh
$ jobs -s
dash: 1: jobs: Illegal option -s
$ jobs -r
dash: 2: jobs: Illegal option -r
```

I would expect shellcheck to warn under SC2039: "In POSIX sh, jobs flags beside -l and -p are undefined".

If you agree with the issue, I can submit a pull request adding the check.

--- a/src/ShellCheck/Checks/ShellSupport.hs
+++ b/src/ShellCheck/Checks/ShellSupport.hs
@@ -138,6 +138,10 @@ prop_checkBashisms55= verify checkBashisms "#!/bin/sh\necho ${@%foo}"
 prop_checkBashisms56= verifyNot checkBashisms "#!/bin/sh\necho ${##}"
 prop_checkBashisms57= verifyNot checkBashisms "#!/bin/dash\nulimit -c 0"
 prop_checkBashisms58= verify checkBashisms "#!/bin/sh\nulimit -c 0"
+prop_checkBashisms59 = verify checkBashisms "#!/bin/sh\njobs -s"
+prop_checkBashisms60 = verifyNot checkBashisms "#!/bin/sh\njobs -p"
+prop_checkBashisms61 = verifyNot checkBashisms "#!/bin/sh\njobs -lp"
+
 checkBashisms = ForShell [Sh, Dash] $ \t -> do
     params <- ask
     kludge params t
@@ -283,6 +287,7 @@ checkBashisms = ForShell [Sh, Dash] $ \t -> do
         allowedFlags = Map.fromList [
             ("exec", []),
             ("export", ["-p"]),
+            ("jobs", ["l", "p"]),
             ("printf", []),
             ("read", if isDash then ["r", "p"] else ["r"]),
             ("ulimit", if isDash then ["H", "S", "t", "f", "d", "s", "c", "m", "l", "p", "n"] else ["f"])

